ngraph/core/reference/include/ngraph/runtime/reference/ctc_greedy_decoder_seq_len.hpp

```diff
@@ -0,0 +1,73 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <algorithm>
+#include <limits>
+#include <vector>
+#include "ngraph/coordinate_transform.hpp"
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            template <typename TF, typename TI, typename TCI, typename TSL>
+            void ctc_greedy_decoder_seq_len(const TF* data,
+                                            const TI* sequence_length,
+                                            const TI* blank_index,
+                                            TCI* out1,
+                                            TSL* out2,
+                                            const Shape& data_shape,
+                                            const Shape& out_shape,
+                                            const bool ctc_merge_repeated)
+            {
+                const auto batch_size = data_shape[0];
+                const auto class_count = data_shape[2];
+
+                CoordinateTransform out_transform = CoordinateTransform(out_shape);
+                CoordinateTransform data_transform = CoordinateTransform(data_shape);
+
+                std::vector<TCI> tmp_out(shape_size(out_shape), -1);
+
+                for (unsigned int batch_ind = 0; batch_ind < batch_size; batch_ind++)
+                {
+                    TI previous_class_index = static_cast<TI>(-1);
```

Comment:
`<limits>` are included so have you consider switch this to `std::numeric_limits<TI>::max()` instead `-1`.